scripts/todo_scanner.py

```
@@ -0,0 +1,459 @@
+    for pattern in EXCLUDE_PATTERNS:
+        if re.search(pattern, path):
+            return True
+    if not include_tests:
+        for pattern in TEST_PATTERNS:
+            if re.search(pattern, path):
+                return True
+    return False
+
+
+def list_repo_files(root: Path) -> list[Path]:
+    result = subprocess.run(
```

Ruff (0.14.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)